www/caddy-custom/Makefile

```
@@ -34,14 +34,17 @@ XCADDY_VERSION!=${MAKE} -C ${.CURDIR}/../xcaddy -V '$${DISTVERSION}.$${PORTREVIS
 PLUGINS+=	--with ${plugin}
 .endfor
 
+# Extract the first three numbers from CADDY_VERSION and prepend with 'v'
+BUILDVERSION= v${CADDY_VERSION:C/^([0-9]+)\.([0-9]+)\.([0-9]+).*/\1.\2.\3/}
```

```suggestion
CADDY_BUILD!=	${MAKE} -C ${.CURDIR}/../caddy  -v '$${DISTVERSIONPREFIX}$${DISTVERSION}'
```

I'd go straight for this like the Makefile already does. Note that -V is now -v which should be replaced in the other calls above too and all of it clustered together.

---

I don't even know if $${PORTREVISION} makes any sense if we just steal the version number from www/caddy and don't do anything else with it.

---

I don't understand what the CADDY_BUILD! variable is doing. What will it contain?

It has to contain an exact github tagged version, like `v2.7.6` and pass it to `xcaddy build v2.7.6`

My current solution does exactly that.

---

Yes, but you recompute information already available... look at www/caddy/Makefile DISTVERSIONPREFIX and DISTVERSION which is what you want. We should aim to extend the port in the way the author thought it would make sene so it's easier to get it accepted.

---

Ohhhh yes that makes sense! I didn't see that.

---

I don't know about PORTREVISION either but lets just keep it the way it is to make the change we want more clear in this PR.

Also the build with your suggestion works just fine so I commited it, thank you!